src/Agent.Plugins/Artifact/PipelineArtifactServer.cs

```diff
@@ -306,13 +310,17 @@ internal async Task DownloadAsyncV2(
 
                 ArtifactProviderFactory factory = new ArtifactProviderFactory(context, connection, this.tracer);
                 IArtifactProvider provider = factory.GetProvider(buildArtifact);
+                ResourceTypes.Add(buildArtifact.Resource.Type);
 
                 await provider.DownloadSingleArtifactAsync(downloadParameters, buildArtifact, cancellationToken, context);
             }
             else
             {
                 throw new InvalidOperationException($"Invalid {nameof(downloadOptions)}!");
             }
+
+            // Create a variable to store the resource types of the downloaded artifacts
+            context.SetVariable("ResourceTypes", string.Join(",", ResourceTypes));
```

Comment:
ResourceTypes is pretty generic, could conflict with customers use of same name?